### ISSUE TITLE
Comment indentation is not kept if preceding an empty block

### ISSUE THREAD

**Issue by jrm2k6** [eemeli/yaml#374]
**Describe the bug**
When I have the follow YAML:

```
---
test1:

# Comment
test2:

# Comment2
test3:
  - random_value
```

**To Reproduce**
```
const document = YAML.parseDocument(
  `
---
test1:

# Comment
test2:

# Comment2
test3:
  - random_value
`
)

console.log(document.toString())

```

It outputs:

```
---
test1:
 # Comment
test2:
 # Comment2
test3:
  - random_value
```

**Expected behaviour**
I would expect the comments to have no indentation, as it is the case in the original document. It looks like an empty block causes that issue. The YAML document seems to be a valid YAML document based on the linting I ran on it.

**Versions (please complete the following information):**

Environment: 
- node: v14.18.2
- `yaml`: [2.0.0-11]

**Comment by eemeli** [comment]
This was pretty interesting; should be now fixed on master. Effectively, the problem here is that for an empty node you can't tell that a comment doesn't belong to the node until you've found the actual next node. So there's now some special-casing around this, moving the association of empty (comment/space/newline) nodes after a blank line, if the value is also empty.

**Comment by jrm2k6** [comment]
@eemeli Thanks for fixing! Is there a new release planned soon including those last commits? Thanks!

**Comment by eemeli** [comment]
Probably not really right away, as I just released `2.0.0-11` a few days ago. Need to figure out what's really still required before I feel comfortable releasing the non-prerelease `2.0.0`.
